src/game/UIStateBar.cpp

```diff
@@ -245,11 +295,45 @@ void CUIStateBar::UpdatePosition(const __Vector3 & vPos, float fYaw) {
     m_fYawPlayer = fYaw;
 }
 
+void CUIStateBar::UpdateFPS(float fps) {
+    if (NULL == m_pText_Fps) {
+        return;
+    }
+
+    char szPos[64];
+    sprintf(szPos, "%.0f", fps);
+    m_pText_Fps->SetString(szPos);
+}
```

Comment:
This can be removed and inlined in Tick IMHO.
```cpp
m_fFPSValue += s_fSecPerFrm;
if (m_fFPSValue > 1.0f) {
    char szBuff[12]{};
    sprintf(szBuff, "%.1f", s_fFrmPerSec);
    m_pText_FPS->SetString(szBuff);
    m_fFPSValue = 0.0f;
}
```

Also note the C++11 initializer list operator `{}`. If you don't do that, some junk data may be output, but won't be the case here, since it's fixed size.